src/category_theory/preadditive/injective.lean

```diff
@@ -301,5 +301,68 @@ by convert congr_arg quiver.hom.unop
 end
 
 end injective
+namespace adjunction
 
+variables {D : Type*} [category D] {F : C ⥤ D} {G : D ⥤ C}
+
+lemma map_injective (adj : F ⊣ G) [F.preserves_monomorphisms] (I : D) (hI : injective I) :
+  injective (G.obj I) :=
+begin
+  constructor,
+  intros X Y f g hg,
+  haveI : mono (F.map g) := by unfreezingI { apply_instance },
+  rcases @hI.1 (F.map f ≫ adj.counit.app _) (F.map g),
+  use adj.unit.app Y ≫ G.map w,
+  rw [←unit_naturality_assoc, ←G.map_comp, h],
+  simp,
+end
+
+lemma injective_of_map_injective (adj : F ⊣ G) [full G] [faithful G] (I : D)
+  (hI : injective (G.obj I)) : injective I :=
+begin
+  constructor,
+  intros X Y f g hg,
+  haveI := adj.right_adjoint_preserves_limits,
+  haveI : mono (G.map g) := by unfreezingI { apply_instance },
+  rcases @hI.1 (G.map f) (G.map g),
+  use inv (adj.counit.app _) ≫ F.map w ≫ adj.counit.app _,
+  refine faithful.map_injective G _,
+  simpa
+end
+
+/-- Given an adjunction `F ⊣ G` such that `F` preserves monos, `G` maps an injective presentation
+of `X` to an injective presentation of `G(X)`. -/
+def map_injective_presentation (adj : F ⊣ G) [F.preserves_monomorphisms] (X : D)
+  (I : injective_presentation X) : injective_presentation (G.obj X) :=
+{ J := G.obj I.J,
+  injective := adj.map_injective _ I.injective,
+  f := G.map I.f,
+  mono := by haveI := I.mono; haveI := adj.right_adjoint_preserves_limits; apply_instance }
```

Comment:
If you add `attribute [instance] injective_presentation.injective injective_presentation.mono` after the definition of the structure `injective_presentation`, the `mono` instance could be infered automatically:
```suggestion
  mono := by haveI := adj.right_adjoint_preserves_limits; apply_instance }
```
(Similarly for `map_projective_presentation`.)